activation/atxdb.go

```diff
@@ -659,6 +648,37 @@ func (db *DB) GetPosAtxID() (types.ATXID, error) {
 	return idAndLayer.AtxID, nil
 }
 
+// addToEpochWeight increments the total weight of epochID by additionalWeight. It is thread-safe.
+func (db *DB) addToEpochWeight(epochID types.EpochID, additionalWeight uint64) error {
+	db.epochWeightMutex.Lock()
+	defer db.epochWeightMutex.Unlock()
+
+	weight, err := db.GetEpochWeight(epochID)
+	if err != nil && err != database.ErrNotFound {
+		return fmt.Errorf("failed to get previous weight: %v", err)
+	}
+	newWeight := weight + additionalWeight
+	if newWeight < weight {
+		newWeight -= newWeight + 1 // this under-flows newWeight so it holds the max possible value
```

Comment:
You are basically reducing the total weight here. Wouldn't it create discrepancies in the eligibility proofs? It's not clear if the current implementation is equipped to handle this scenario.